docs/test-development.md

```diff
@@ -0,0 +1,43 @@
+## Writing tests
+
+We focus primarily on integration/functional level tests. Unit tests are avoided and only recommended for isolated
+classes such as small utils. We do not use any mocking frameworks and we will not accept any contributions that adds a 
+mocking framework.
+
+When writing tests please follow the same approach as we have taken in the other tests. There are many ways to 
+test software and we have chosen ours, so please appreciate that.
+
+The main tests are provided in `src/test/java` for each Quickstart folder. Most of the integration tests are there.
+
+Any test inside the quickstarts should be reasonable and straightforward to understand. But feedback for improvements are always welcome.
+
+When developing your test depending on the feature or enhancement you are testing you may find it best to add to an
+existing test, or to write a test from scratch. For the latter, we recommend finding another test that is close to what 
+you need and use that as a basis.
+
+
+Running integration tests
+-------------------------
```

Comment:
I believe we should use markdown syntax for headers (something like `## Running integration tests`)

Reply:
Agreed and updated.